File: app/src/main/java/com/sagar/watchnext/network/models/people/KnownFor.java

package com.sagar.watchnext.network.models.people;

import java.util.List;

import com.google.gson.annotations.SerializedName;

public class KnownFor {

    @SerializedName("poster_path")
    private String posterPath;

    @SerializedName("adult")
    private boolean adult;

    @SerializedName("overview")
    private String overview;

    @SerializedName("release_date")
    private String releaseDate;

    @SerializedName("original_title")
    private String originalTitle;

    @SerializedName("genre_ids")
    private List<Integer> genreIds;

    @SerializedName("id")
    private int id;

    @SerializedName("media_type")
    private String mediaType;

    @SerializedName("original_language")
    private String originalLanguage;

    @SerializedName("title")
    private String title;

    @SerializedName("backdrop_path")
    private String backdropPath;

    @SerializedName("popularity")
    private float popularity;

    @SerializedName("vote_count")
    private int voteCount;

    @SerializedName("video")
    private boolean video;

    @SerializedName("vote_average")
    private float voteAverage;

    @SerializedName("first_air_date")
    private String firstAirDate;

    @SerializedName("origin_country")
    private List<String> originCountry;

    @SerializedName("name")
    private String name;

    @SerializedName("original_name")
    private String originalName;

    public String getPosterPath() {
        return posterPath;
    }

    public boolean isAdult() {
        return adult;
    }

    public String getOverview() {
        return overview;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public String getOriginalTitle() {
        return originalTitle;
    }

    public List<Integer> getGenreIds() {
        return genreIds;
    }

    public int getId() {
        return id;
    }

    public String getMediaType() {
        return mediaType;
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public String getTitle() {
        return title;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public float getPopularity() {
        return popularity;
    }

    public int getVoteCount() {
        return voteCount;
    }

    public boolean isVideo() {
        return video;
    }

    public float getVoteAverage() {
        return voteAverage;
    }

    public String getFirstAirDate() {
        return firstAirDate;
    }

    public List<String> getOriginCountry() {
        return originCountry;
    }

    public String getName() {
        return name;
    }

    public String getOriginalName() {
        return originalName;
    }
}

File: app/src/main/java/com/sagar/watchnext/network/models/people/KnownFor.kt
package com.sagar.watchnext.network.models.people

import com.google.gson.annotations.SerializedName

class KnownFor {

    @SerializedName("poster_path")
    val posterPath: String? = null

    @SerializedName("adult")
    val isAdult: Boolean = false

    @SerializedName("overview")
    val overview: String? = null

    @SerializedName("release_date")
    val releaseDate: String? = null

    @SerializedName("original_title")
    val originalTitle: String? = null

    @SerializedName("genre_ids")
    val genreIds: List<Int>? = null

    @SerializedName("id")
    val id: Int = 0

    @SerializedName("media_type")
    val mediaType: String? = null

    @SerializedName("original_language")
    val originalLanguage: String? = null

    @SerializedName("title")
    val title: String? = null

    @SerializedName("backdrop_path")
    val backdropPath: String? = null

    @SerializedName("popularity")
    val popularity: Float = 0.toFloat()

    @SerializedName("vote_count")
    val voteCount: Int = 0

    @SerializedName("video")
    val isVideo: Boolean = false

    @SerializedName("vote_average")
    val voteAverage: Float = 0.toFloat()

    @SerializedName("first_air_date")
    val firstAirDate: String? = null

    @SerializedName("origin_country")
    val originCountry: List<String>? = null

    @SerializedName("name")
    val name: String? = null

    @SerializedName("original_name")
    val originalName: String? = null
}
